lib/std/mem/Allocator.zig

```diff
@@ -297,7 +297,7 @@ pub fn free(self: Allocator, memory: anytype) void {
     const bytes = mem.sliceAsBytes(memory);
     const bytes_len = bytes.len + if (Slice.sentinel != null) @sizeOf(Slice.child) else 0;
     if (bytes_len == 0) return;
-    const non_const_ptr = @intToPtr([*]u8, @ptrToInt(bytes.ptr));
+    const non_const_ptr = @qualCast([*]u8, @as([*]const u8, bytes.ptr));
```

Comment:
The fact that this @as() is required is a bit awkward. Without it I get
```
lib/std/mem/Allocator.zig:300:27: error: '@qualCast' can only modify 'const' and 'volatile' qualifiers
    const non_const_ptr = @qualCast([*]u8, bytes.ptr);
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
lib/std/mem/Allocator.zig:300:27: note: expected type '[*]u8'
lib/std/mem/Allocator.zig:300:27: note: got type '[*]align(8) u8'
```
However, `[*]align(8) u8` coerces to the expected type of `[*]u8` so it seems to me like this explicit coercion with `@as()` shouldn't be necessary.